lib/cc/engine/analyzers/issue.rb

```diff
@@ -0,0 +1,89 @@
+require "cc/engine/analyzers/sexp_lines"
+require "cc/engine/analyzers/violation_read_up"
+require "digest"
+
+module CC
+  module Engine
+    module Analyzers
+      class Issue
+        def initialize(language_strategy:, check_name:, current_sexp:, other_sexps:)
+          @language_strategy = language_strategy
+          @check_name = check_name
+          @current_sexp = current_sexp
+          @other_sexps = other_sexps
+        end
+
+        def format # rubocop:disable Metrics/MethodLength
+          {
+            "type": "issue",
+            "check_name": check_name,
+            "description": description,
+            "categories": ["Duplication"],
+            "location": format_location,
+            "remediation_points": calculate_points,
+            "other_locations": format_other_locations,
+            "content": content_body,
+            "fingerprint": fingerprint,
+          }
+        end # rubocop:enable Metrics/MethodLength
+
+        def report_name
+          "#{current_sexp.file}-#{current_sexp.line}"
+        end
+
+        def mass
+          current_sexp.mass
+        end
+
+        private
+
+        attr_reader :language_strategy, :check_name, :other_sexps, :current_sexp
+
+        def calculate_points
+          language_strategy.calculate_points(mass)
+        end
+
+        def format_location
+          format_sexp(current_sexp)
+        end
+
+        def format_other_locations
+          other_sexps.map do |sexp|
+            format_sexp(sexp)
+          end
+        end
+
+        def format_sexp(sexp)
+          lines = SexpLines.new(sexp)
+          {
+            "path": sexp.file.gsub(%r{^./}, ""),
+            "lines": {
+              "begin": lines.begin_line,
+              "end": lines.end_line,
+            },
+          }
+        end
+
+        def content_body
+          { "body": ViolationReadUp.new(mass).contents }
+        end
+
+        def fingerprint
+          digest = Digest::MD5.new
+          digest << current_sexp.file
```

Comment:
Ha, No worries. :)

This is the code I found in classic: https://github.com/codeclimate/analyzer/blame/master/lib/quality/smells/duplication.rb

It uses the constant name, so the matching fingerprints would only be for duplication issues within the same file. Looking at the commit history, it's not completely clear to me why that was done - it doesn't seem to have any impact now in the UI- and I agree that it's a bit nonsensical for reporting new issues 

`IssuesDelta` In Finalizer: https://github.com/codeclimate/finalizer/blob/master/lib/cc/finalizer/issues_delta.rb#L21